portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java

```diff
@@ -0,0 +1,104 @@
+package com.google.sps.data;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+public class LongestTaskFirstScheduler implements TaskScheduler {
+
+  // Comparator for sorting tasks by duration in descending order and then by task name
+  // alphabetically.
+  public static final Comparator<Task> sortByTaskDurationDescendingThenName =
```

Comment:
You may want to use the task priority as the second sort param, rather than name

Reply:
Updated. I also added a `compareTo` method in TaskPriority.java